service/controller/rbac/resource/fluxauth/delete.go

```diff
@@ -54,7 +54,7 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 	}
 
 	clusterRoleBindings := []string{
-		pkgkey.WriteSilencesAutomationSARoleBindingName(),
+		pkgkey.WriteSilencesAutomationSAinNSRoleBindingName(ns.Name),
```

Comment:
When the namespace is deleted we only delete the clusterrolebinding referencing it. This prevents deletion of bindings for other SAs that are still needed.